dev_cmds/integration-tests/single-package-json-example.ts

```diff
@@ -64,9 +63,32 @@ export function createSinglePackageJsonExampleTestGroup(): TestGroup {
     }
   };
 
-  const runFailsWithErrorCmd: TestFunction = async (containerName: string) => {
-    const exampleCmdCommandLine = `npx devcmd fails_with_error`;
+  const runExampleCmdWithExtension: TestFunction = async (containerName: string) => {
+    const { stdout, stderr } = await execToString({
+      command: DOCKER_COMMAND,
+      args: [
+        "exec",
+        containerName,
+        "sh",
+        "-c",
+        ["cd /tmp/devcmd_test/single-package-json", `npx devcmd example_cmd.ts`].join(" && "),
+      ],
+    });
 
+    if (!stdout.includes("Example command for single-package-json example")) {
```

Comment:
I know your PR isn't really the problem here, but maybe we should think about extracting this logging logic into a util function, since it is copied in almost every error case in each integration test 🤔 I can imagine something like this

```typescript
function logStdoutFailure(actual: string, expected: string)
```

would reduce code duplication in this project. Maybe it would even be a good idea to extract line 78 to 88 entirely ^^

Reply:
on the one hand, I agree we shouldn't let the duplication here increase too much.
on the other hand, I'd also prefer not to build our own testing framework from scratch :D

I think with the exit code handling in #48 and the still-to-do "suppressNotices" option (that I explained here: https://github.com/XITASO/devcmd/issues/47#issuecomment-1001685196), we should re-evaluate whether we could use a regular testing framework (like Jest) for our integration tests instead.

I'll create a new issue about this.